feat: add inpainting support to Amazon Nova Canvas API

SantoshDhaladhuli opened the pull request:
Title
feat: add inpainting support to Amazon Nova Canvas API
Pre-Submission checklist
 I have added testing in the [tests/test_litellm/llms/bedrock/image/test_amazon_nova_canvas_transformation.py](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, Adding at least 1 test is a hard requirement – [see details](https://docs.litellm.ai/docs/extras/contributing_code)
 I have added a screenshot of my new test passing locally
 My PR passes all unit tests on [make test-unit](https://docs.litellm.ai/docs/extras/contributing_code)
 My PR's scope is as isolated as possible – it only solves 1 specific problem
Type
🆕 New Feature
✅ Test
Changes
Added support for inpainting with Amazon Nova Canvas API in the LiteLLM provider module
Implemented request formatting and response parsing logic for the new inpainting use-case
Added unit test to validate Amazon Nova Canvas inpainting functionality

Commit:
feat: add inpainting support to Amazon Nova Canvas API

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/BerriAI/litellm?pullRequest=12755) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/BerriAI/litellm?pullRequest=12755) before we can accept your contribution.<br/><hr/>**Sai Santosh** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/BerriAI/litellm?pullRequest=12755) it.</sub>

SantoshDhaladhuli commented:
I've signed the CLA, please let me know if you got it or not.

Commit:
fixed bug on type in params for Amazon Nova Canvas inpainting
# This file is part of LiteLLM.

SantoshDhaladhuli commented:
<img width="1211" height="931" alt="Screenshot 2025-07-19 at 9 45 49 PM" src="https://github.com/user-attachments/assets/cc9e27b3-79a1-4751-9cb7-eb5a72cc6007" />
Please check the screenshot for reference,.

Also, All checks have passed, you can test as well.

ishaan-jaff committed:
bump litellm enterprise version

colesmcintosh committed:
feat: add Hyperbolic provider support

- Add Hyperbolic as a new OpenAI-compatible provider
- Implement HyperbolicChatConfig inheriting from OpenAILikeChatConfig
- Register Hyperbolic in provider lists and constants
- Add comprehensive model configurations with pricing for:
  - DeepSeek models (V3, R1, etc.)
  - Qwen models (2.5, 3, QwQ, etc.)
  - Meta Llama models (3.1, 3.2, 3.3)
  - Other models like Kimi K2, Hermes 3, etc.
- Configure default API base URL: https://api.hyperbolic.xyz/v1
- Add provider documentation with usage examples
- Create unit tests for provider functionality
- Support all standard OpenAI parameters

Hyperbolic provides low-cost inference with OpenAI-compatible APIs,
supporting latest models without infrastructure overhead.

ishaan-jaff committed:
[Feat] Add Recraft Image Generation API Support - New LLM Provider (#12832)

* add recraft

* init RecraftImageGenerationConfig

* add get_complete_url + validate_environment

* add image_generation_handler in llm http clas

* fixes for transform

* working recraft request

* fixed img gen transform

* fixes for llm http handler

* test: TestRecraftImageGeneration

* fixes for llm_http_handler

* fix RecraftImageGenerationConfig

* TestRecraftImageGenerationTransformation

* add recraft API

* docs recraft API

* fix code QA

* map_openai_params

* fix recraft

* cost tracking for recraft/recraftv3

* fix code qa check

Commit:
Litellm dev 07 21 2025 p1 (#12848)

* fix(main.py): fix async retryer

Fixes https://github.com/BerriAI/litellm/issues/12830

* fix(forward_clientside_headers_by_model_group.py): filter out 'content-type' from forwardable headers

clientside content-type != proxy content type, can cause requests to hang

* test(tests/): update tests

ishaan-jaff committed:
Revert "Litellm dev 07 21 2025 p1 (#12848)"

This reverts commit e4e10aa4edcb9b8e69a99b9e166514be2b3c34b4.

SantoshDhaladhuli committed:
Merge branch 'main' into litellm/amazon-nova-inpainting-add

SantoshDhaladhuli committed:
Update poetry.lock file generated by Poetry 1.8.3

SantoshDhaladhuli committed:
Update poetry.lock file generated by Poetry 1.8.3

SantoshDhaladhuli committed:
Merge branch 'BerriAI:main' into litellm/amazon-nova-inpainting-add
